src/core/qgssnappingutils.cpp

```diff
@@ -615,7 +615,7 @@ void QgsSnappingUtils::readConfigFromProject()
        layerIdList.size() != snapToList.size() )
     return;
 
-  if ( !snappingDefinedInProject )
+  if ( layerIdList.isEmpty() || !snappingDefinedInProject || !ok )
```

Comment:
@wonder-sk:
* parsing errors in the settings were just ignored.
* I don't really remember... old PR. I guess to leave the snapMode to SnapCurrentLayer.